spec_plots/utils/specutils_hasp/plotspec.py

```diff
@@ -0,0 +1,347 @@
+    if is_bigplot:
+        this_plotarea.set_title(title_addendum, loc="right", size="small",
+                                color="red")
+
+    # Extract the optimal x-axis plot range from the plot_metrics dict,
+    # since we use it a lot.
+    optimal_xaxis_range = plot_metrics["optimal_xaxis_range"]
+
+    # Plot the spectrum, but only if valid wavelength ranges for x-axis
+    # are returned, otherwise plot a special "Fluxes Are All Zero" plot.
+    if all(numpy.isfinite(optimal_xaxis_range)):
+        # We plot the spectrum as a regular line for use
+        # in calc_covering_fraction, it will be removed later.
+        this_line = this_plotarea.plot(all_wls, all_fls, 'b')
+
+        # Update y-axis range, but only adjust the ranges if this isn't
+        # an all-zero flux case (and not in debug mode, in which case I want
+        # to see the entire y-axis range).
+        if not debug:
+            this_plotarea.set_ylim(plot_metrics["y_axis_range"])
+
+        covering_fraction = calc_covering_fraction(this_figure,
+                                                   these_plotareas, 0,
+                                                   optimize=optimize)
+        # Note: here we remove the line we plotted before, it was only
+        # so that calc_covering_fraction would have someting to draw on the
+        # canvas and thereby determine which pixels were "blue" (i.e., part
+        # of the plotted spectrum vs. background).
+        this_line.pop(0).remove()
+        # Now we plot the spectrum as a LineCollection so that the
+        # transparency will have the desired effect, but, this is not
+        # rendered on the canvas inside calc_covering_fraction, hence why we
+        # need to plot it both as a regular line first.
+        # Note: because we are re-using a LineCollection object in the
+        # array of plot_metrics (specifically, when creating the
+        # thumbnail-sized plot), we have to use a copy of the LineCollection
+        # object, otherwise it will have Axes, Figure, etc. all defined and
+        # resetting them to None does not work.  Since this is only an issue
+        # with thumbnail-sizes, this is only relevant for the first
+        # LineCollection.
+        this_collection = this_plotarea.add_collection(copy.copy(
+            plot_metrics["line_collection"]))
+
+        if covering_fraction > 30.:
+            this_collection.set_alpha(0.1)
+
+        # Turn on plot grid lines.
+        this_plotarea.grid(True, linestyle='dashed')
+
+        if is_bigplot:
+            this_figure.suptitle(os.path.basename(
+                hasp_spectrum.plot_info['orig_file']), fontsize=18, color='r')
+            # Uncomment the lines below to include the covering fraction
+            # as part of the suptitle.
+
+        if debug:
+            # Overplot points color-coded based on rejection criteria.
+            debug_oplot(this_plotarea, "stis", all_wls, all_fls,
```

Comment:
Change stis to hasp. Verify whether updates to debug_oplot are needed.

Reply:
Updated "stis" to "hasp". debug_oplot calls is_bd_dpq() and passes in "hasp". is_bad_dpq() can already handle "hasp".